lib/nx_signal.ex

```diff
@@ -297,4 +467,155 @@ defmodule NxSignal do
 
     output
   end
+
+  @doc """
+  Generates weights for converting an STFT representation into MEL-scale.
+
+  See also: `stft/3`, `istft/3`, `stft_to_mel/2`
+
+  ## Options
+
+    * `:nfft` - Number of FFT bins
+    * `:fs` - Sampling frequency in Hz
+    * `:nmels` - Number of target MEL bins. Defaults to 128
+    * `:max_mel` - the pitch for the last MEL bin before log scaling. Defaults to 3016
+    * `:mel_frequency_spacing` - the distance in Hz between two MEL bins before log scaling. Defaults to 66.6
+    * `:type` - Target output type. Defaults to `{:f, 32}`
+
+  ## Examples
+
+      iex> NxSignal.mel_filters(nfft: 10, fs: 8.0e3, nmels: 5)
+      #Nx.Tensor<
+        f32[5][frequencies: 10]
+        [
+          [0.0, 8.129207417368889e-4, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+          [0.0, 9.972017724066973e-4, 2.1870265481993556e-4, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+          [0.0, 0.0, 9.510896052233875e-4, 4.1505080298520625e-4, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0, 4.035892488900572e-4, 5.276654846966267e-4, 2.5741232093423605e-4, 0.0, 0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0, 0.0, 7.329042273340747e-5, 2.3422071535605937e-4, 3.8295125705190003e-4, 2.8712046332657337e-4, 1.9128969870507717e-4, 9.545891225570813e-5]
+        ]
+      >
+  """
+  defn mel_filters(opts \\ []) do
```

Comment:
You should make this function a `deftransform` and all numeric values that are given as options should be passed as regular arguments to the underlying `defnp mel_filters_impl` :)